network/setup-ip

```diff
@@ -188,36 +196,28 @@ fi
 
 if [ "$ACTION" == "add" ]; then
     MAC="$(get_mac_from_iface "$INTERFACE")"
-    if [ -n "$MAC" ]; then
-        ip="$(/usr/bin/qubesdb-read "/net-config/$MAC/ip" 2> /dev/null)" || ip=
-        ip6="$(/usr/bin/qubesdb-read "/net-config/$MAC/ip6" 2> /dev/null)" || ip6=
-        netmask="$(/usr/bin/qubesdb-read "/net-config/$MAC/netmask" 2> /dev/null)" || netmask=
-        netmask6="$(/usr/bin/qubesdb-read "/net-config/$MAC/netmask6" 2> /dev/null)" || netmask6=
-        gateway="$(/usr/bin/qubesdb-read "/net-config/$MAC/gateway" 2> /dev/null)" || gateway=
-        gateway6="$(/usr/bin/qubesdb-read "/net-config/$MAC/gateway6" 2> /dev/null)" || gateway6=
-
-        # Handle legacy values
-        LEGACY_MAC="$(/usr/bin/qubesdb-read /qubes-mac 2> /dev/null)" || LEGACY_MAC=
-        if [ "$MAC" == "$LEGACY_MAC" ] || [ -z "$LEGACY_MAC" ]; then
-            if [ -z "$ip" ]; then
-                ip="$(/usr/bin/qubesdb-read /qubes-ip 2> /dev/null)" || ip=
-            fi
-            if [ -z "$ip6" ]; then
-                ip6="$(/usr/bin/qubesdb-read /qubes-ip6 2> /dev/null)" || ip6=
-            fi
-            if [ -z "$gateway" ]; then
-                gateway="$(/usr/bin/qubesdb-read /qubes-gateway 2> /dev/null)" || gateway=
-            fi
-            if [ -z "$gateway6" ]; then
-                gateway6="$(/usr/bin/qubesdb-read /qubes-gateway6 2> /dev/null)" || gateway6=
-            fi
-        fi
-
-        if [ -z "$netmask" ]; then
-            netmask="255.255.255.255"
+    prefix="/net-config/$MAC"
+    if [[ -n "$MAC" ]]; then
+        custom=$(exec /usr/bin/qubesdb-exists "${prefix}custom")
+        if ip4=$(exec /usr/bin/qubesdb-read "${prefix}ip" 2>/dev/null); then
+            :
+        elif [[ "$?" != '2' ]]; then
+            echo "Could not check if /net-config/$MAC/ip exists!" >&2
+            exit 1
+        else
+            prefix=/qubes-
+            ip4=$(exec /usr/bin/qubesdb-read "${prefix}ip")
```

Comment:
shellcheck complains this variable is unused, which means you haven't updated its name down below.